tests/cohort_is_member_test.php

```diff
@@ -0,0 +1,163 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Boost navigation fumbling cohort is member PHP unit tests.
+ *
+ * @package   local_boostnavigation
+ * @author    2020 Anupama Dharmajan <anupamadharmajan@catalyst-au.net>
+ * @copyright Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die;
+
+require_once(__DIR__ . '/../locallib.php');
+
+global $CFG;
+
+/**
+ * Unit tests for cohort is member.
+ *
+ * @package   local_boostnavigation
+ * @author    2020 Anupama Dharmajan <anupamadharmajan@catalyst-au.net>
+ * @copyright Catalyst IT
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+class local_boostnavigation_test_cohort_is_member extends advanced_testcase {
+
+    /**
+     * Test user cohorts after cohort add member.
+     */
+    public function test_cohort_add_member() {
+        global $DB;
+
+        $this->resetAfterTest();
+
+        $cohort = $this->getDataGenerator()->create_cohort(array('idnumber' => '01'));
+        $user = $this->getDataGenerator()->create_user();
+
+        // Check user is not a member in cohort.
+        $cohortismember = local_boostnavigation_cohort_is_member($user->id, $cohort->idnumber);
+        $this->assertFalse($cohortismember);
+
+        // Sleep for 1 seconds.
+        sleep(1);
```

Comment:
avoid sleep in unit tests, they are always slow enough without making them slower

If you are working with time based tests always make sure you can pass in the time in the public api of whatever you are dealing with to make it easier to test.

Either way you should not even need to sleep here, what was the intent behind this?